sql/api/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -602,14 +602,25 @@ propertyList
     ;
 
 property
-    : key=propertyKey (EQ? value=propertyValue)?
+    : key=propertyKeyOrStringLit EQ value=propertyValue               #propertyWithKeyAndEquals
+    | key=propertyKeyNoCoalesce value=propertyValue?                  #propertyWithStringKeyNoEquals
+    | key=propertyKey (EQ? value=propertyValue)?                      #propertyWithIdentifierKey
     ;
 
 propertyKey
     : errorCapturingIdentifier (DOT errorCapturingIdentifier)*
+    ;
+
+propertyKeyOrStringLit
+    : propertyKey
     | stringLit
     ;
 
+propertyKeyNoCoalesce
```

Comment:
```suggestion
propertyKeyOrStringLitNoCoalesce
```
It's verbose but avoids any confusion.

Reply:
and since we mention `propertyKey` in the name, shall we match `propertyKey` here as well?